api/controllers/Irc_EntryController.js

```diff
@@ -63,32 +57,74 @@ module.exports = {
   },
 
   process_event: function (request_body) {
-    if (request_body.operation === 'check in') {
-      return this.saveDetainee(request_body)
-        .then(this.saveEvent.bind(this, request_body));
+    switch (request_body.operation) {
+    case Event.OPERATION_CHECK_IN:
+    case Event.OPERATION_CHECK_OUT:
+    case Event.OPERATION_INTER_SITE_TRANSFER:
+    case Event.OPERATION_REINSTATEMENT:
+      return this.saveEvent(request_body);
+    default:
+      throw new ValidationError('Invalid operation');
     }
-    throw new ValidationError('Unknown');
   },
 
-  getPID: (entity) => `${entity.centre}_${entity.person_id}`,
+  createOrUpdateDetainee: function (request_body) {
+    return Centres.findOne({ name: request_body.centre })
+      .populate('detainees', { person_id: request_body.person_id })
+      .then(function (centre) {
+        if (!centre) {
+          throw new Error('Centre could not be found');
+        }
+        if (centre.detainees.length === 1) {
+          var detainee = centre.detainees[0];
+          if (!detainee) {
+            throw new Error('Detainee could not be found');
```

Comment:
what scenario are you catching with this, that wouldn't have been caught when you got the `centre.detainees.length === 1` ?